speechbrain/utils/evaluate_dialogue_state_tracking.py

```diff
@@ -0,0 +1,685 @@
+
+multiwoz_slot_types = [
+    "attraction-area",
+    "attraction-name",
+    "attraction-type",
+    "hotel-area",
+    "hotel-name",
+    "hotel-type",
+    "hotel-day",
+    "hotel-people",
+    "hotel-pricerange",
+    "hotel-stay",
+    "hotel-stars",
+    "hotel-internet",
+    "hotel-parking",
+    "restaurant-area",
+    "restaurant-name",
+    "restaurant-food",
+    "restaurant-day",
+    "restaurant-people",
+    "restaurant-pricerange",
+    "restaurant-time",
+    "taxi-arriveby",
+    "taxi-departure",
+    "taxi-destination",
+    "taxi-leaveat",
+    "train-arriveby",
+    "train-departure",
+    "train-destination",
+    "train-leaveat",
+    "train-people",
+    "train-day",
+    "hospital-department",
+    "bus-people",
+    "bus-leaveat",
+    "bus-arriveby",
+    "bus-day",
+    "bus-destination",
+    "bus-departure",
+]
+
+multiwoz_time_slots = [
+    "restaurant-time",
+    "taxi-arriveby",
+    "taxi-leaveat",
+    "train-arriveby",
+    "train-leaveat",
+]
+
+multiwoz_open_slots = [
+    "attraction-name",
+    "hotel-name",
+    "restaurant-name",
+    "taxi-departure",
+    "taxi-destination",
+    "train-departure",
+    "train-destination",
+]
+
+
+def dialogueState_str2dict(
+    dialogue_state: str, slot_type_filtering: list[str] = None
+):
+    """
+    Converts the ; separated Dialogue State linearization to a domain-slot-value dictionary.
+    When *slot_type_filtering* is provided, it filters out the slots which are not part of this list.
+    """
+    dict_state = {}
+
+    # We consider every word after "[State] " to discard the transcription if present.
+    dialogue_state = dialogue_state.split("[State] ")[-1]
+    if ";" not in dialogue_state:
+        return {}
+    else:
+        slots = dialogue_state.split(";")
+        for slot_value in slots:
+            if "=" not in slot_value:
+                continue
+            else:
+                slot, value = (
+                    slot_value.split("=")[0].strip(),
+                    slot_value.split("=")[1].strip(),
+                )
+                if slot_type_filtering and slot not in slot_type_filtering:
+                    continue
+                elif "-" in slot:
+                    domain, slot_type = (
+                        slot.split("-")[0].strip(),
+                        slot.split("-")[1].strip(),
+                    )
+                    if domain not in dict_state.keys():
+                        dict_state[domain] = {}
+                    dict_state[domain][slot_type] = value
+
+        return dict_state
+
+
+def dialogueState_dict2str(dialogue_state: dict):
+    """
+    Converts a dialogue state as a dictionary per slot type per domain to a serialized ; separated list
+    """
+    slots = []
+    for domain, slots_values in dialogue_state.items():
+        slots.extend(
+            [f"{domain}-{slot}={value}" for slot, value in slots_values.items()]
+        )
+
+    return "; ".join(slots)
+
+
+class JGATracker:
```

Comment:
Should be moved to `metric_stats.py` and made a child class of `MetricStats` (see the WER stats class for reference).